update_ca.F90

```diff
@@ -1,123 +1,323 @@
-logical,intent(in) :: nca_plumes
-type(domain2D), intent(inout) :: domain_for_coupler
-real, dimension(nlon,nlat) :: frac
-integer, dimension(nlon,nlat) :: maxlives
-integer,allocatable,save :: board(:,:,:), lives(:,:,:)
-integer,allocatable :: V(:),L(:),B(:)
-integer,allocatable :: AG(:,:)
-integer :: inci, incj, i, j, k,sub,spinup,it,halo,k_in,isize,jsize
-integer :: ih, jh,kend
-real, allocatable :: field_in(:,:),board_halo(:,:,:)
-integer, dimension(nxc,nyc) :: neighbours, birth, newlives,thresh,maxliveshigh
+integer, intent(in)  :: nlives,nseed, nspinup, nf,ncells
+real,    intent(in)  :: nfracseed
+logical, intent(in)  :: nca_plumes,restart,first_flag,first_time_step
+type(domain2D), intent(inout) :: domain_ncellx
+real,    dimension(nlon,nlat) :: frac
+integer, allocatable  :: V(:),L(:),B(:)
+integer, allocatable  :: AG(:,:)
+integer              :: inci, incj, i, j, k,sub,spinup,it,halo,k_in,isize,jsize
+integer              :: ih, jh,kend, count4,boardmax,livemax,ilivemax
+real,    allocatable :: board_halo(:,:,:)
+integer, dimension(nxc,nyc) :: neighbours, birth, newlives,thresh
 integer, dimension(nxc,nyc) :: neg, newcell, oldlives, newval,temp,newseed
 integer, dimension(ncells,ncells) :: onegrid
+integer(8)           :: count, count_rate, count_max, count_trunc
+integer(8)           :: iscale = 10000000000
+logical, save        :: start_from_restart
 
 real, dimension(nxc,nyc) :: NOISE_B
 real, dimension(nxc*nyc) :: noise1D2
 
 
+!------------------------------------------------------------------------------------------------
+
+if(first_time_step)then
+start_from_restart = .False.
+endif
+
 !-------------------------------------------------------------------------------------------------
 halo=1
 isize=nlon+2*halo
 jsize=nlat+2*halo
 k_in=1
-
- if (.not. allocated(board))then
- allocate(board(nxc,nyc,nca))
- endif
- if (.not. allocated(lives))then
- allocate(lives(nxc,nyc,nca))
- endif
- if (.not. allocated(field_in))then
- allocate(field_in(nxc*nyc,1))
- endif
- if(.not. allocated(board_halo))then
- allocate(board_halo(nxch,nych,1))   
+ 
+  if (.not. allocated(board))then
+     allocate(board(nxc,nyc,nca))
+  endif
+  if (.not. allocated(lives))then
+     allocate(lives(nxc,nyc,nca))
+  endif
+  if(.not. allocated(board_halo))then
+     allocate(board_halo(nxch,nych,1))
+  endif
+ 
+ !Step 1: Generate a new random number each time-step for "random seeding"
+ !each nseed timestep where mod(kstep,nseed) = 0
+
+ if (iseed_ca == 0) then
+    ! generate a random seed from system clock and ens member number
+    call system_clock(count, count_rate, count_max)
+    ! iseed is elapsed time since unix epoch began (secs)
+    ! truncate to 4 byte integer
+    count_trunc = iscale*(count/iscale)
+    count4 = count - count_trunc
+ else
+    ! don't rely on compiler to truncate integer(8) to integer(4) on
+    ! overflow, do wrap around explicitly.
+    count4 = mod(mype + iseed_ca + 2147483648, 4294967296) - 2147483648
```

Comment:
It looks like the seed is set to the same value each time-step.  Should kdt be added to the see so it would evolve over the forecast?

Reply:
@pjpegion  I did a quick test with kdt in the random seed generation and it looks reasonable, differences are random, and organization looks similar so I will update my PR with this addition, thanks!

Reply:
@pjpegion kdt has been added, thanks